app/components/CredentialCard.tsx

```diff
@@ -22,39 +22,68 @@ import TimeAgo from './TimeAgo'
 import { ToolbarButtonGroup } from './ToolbarButtonGroup'
 import type { RedactedClientCredential } from '~/routes/user.credentials/client_credentials.server'
 
+const WARNING_DAYS = 21 * 24 * 60 * 60 * 1000 // 21 days in milliseconds
+const EXPIRATION_DAYS = 30 * 24 * 60 * 60 * 1000
+
 export default function CredentialCard({
   name,
   client_id,
   created,
   scope,
   scopeDescription,
   lastUsed,
+  expired,
 }: RedactedClientCredential & { scopeDescription?: string }) {
   const ref = useRef<ModalRef>(null)
   const fetcher = useFetcher()
   const [searchParams] = useSearchParams()
   const disabled = fetcher.state !== 'idle'
   const alerts = searchParams.getAll('alerts')
+  const showExpirationWarning = !(
+    expired || Date.now() - (lastUsed ?? 0) <= WARNING_DAYS
+  )
+
   return (
     <>
       <Grid row style={disabled ? { opacity: '50%' } : undefined}>
-        <div className="tablet:grid-col flex-fill">
+        <div
+          className="tablet:grid-col flex-fill"
+          style={expired ? { opacity: '50%' } : undefined}
+        >
           <div>
             <small>
               <strong>{name}</strong>{' '}
               <span>
                 (created <TimeAgo time={created} />,{' '}
-                {lastUsed ? (
+                {expired ? (
                   <>
-                    last used <TimeAgo time={lastUsed} />
+                    expired <TimeAgo time={expired} />
                   </>
                 ) : (
-                  <>never used</>
+                  <>
+                    {lastUsed ? (
+                      <>
+                        last used <TimeAgo time={lastUsed} />
+                      </>
+                    ) : (
+                      <>never used</>
+                    )}
+                  </>
                 )}
                 )
               </span>
             </small>
           </div>
+          {showExpirationWarning && (
+            <small className="text-error">
+              This credential will expire
+              {lastUsed
+                ? ` on ${new Date(lastUsed + EXPIRATION_DAYS).toDateString()} `
+                : ' today '}
```

Comment:
Can we display this English text and a precise date tooltip, like we do in the TimeAgo component?